### DISCUSSION TITLE
DISKANN index type not supported on DataType.FLOAT_VECTOR with IP metric?

### DISCUSSION THREAD

**Discussion by yushangdi** [milvus-io/milvus#30199] [Q&A and General discussion] [answered]
I'm trying to build index using DiskANN type, but got error index type not supported. Here's my schema, I'm wondering why it is not supported? It's building on a float vector with inner product (IP) metric, based on https://milvus.io/docs/disk_index.md, it should be supported?

    fields = [
        FieldSchema(
            name="id",
            dtype=DataType.INT64,
            is_primary=True,
            auto_id=False,
            max_length=100,
        ),
        FieldSchema(name="priority", dtype=DataType.FLOAT),
        FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR, dim=dim), ## dim = 100
    ]
    schema = CollectionSchema(fields, "points with id and priority")
    points = Collection("points", schema, consistency_level="Strong")
    points.insert(entities)
    points.flush()

    index_params: {
        "index_type": "DISKANN",
        "metric_type": "IP",
        "params": {},
    }

     points.create_index("embeddings", index_params)

**Comment by yushangdi** [comment]
Full error stack:

Traceback (most recent call last):
  File "run_milvus.py", line 164, in <module>
    points.create_index("embeddings", index_params)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/orm/collection.py", line 1208, in create_index
    return conn.create_index(self._name, field_name, index_params, timeout=timeout, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/decorators.py", line 129, in handler
    raise e from e
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/decorators.py", line 125, in handler
    return func(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/decorators.py", line 164, in handler
    return func(self, *args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/decorators.py", line 104, in handler
    raise e from e
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/decorators.py", line 68, in handler
    return func(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 842, in create_index
    check_status(status)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pymilvus/client/utils.py", line 54, in check_status
    raise MilvusException(status.code, status.reason, status.error_code)
pymilvus.exceptions.MilvusException: <MilvusException: (code=65535, message=index type not supported[indexType=DISKANN])>

**Comment by yhmo** [comment]
By milvus v2.3.4 + pymilvus v2.3.5, I didn't encounter any error. Make sure your pymilvus and milvus version are v2.3

I use the same script and succeed:
```
fields = [
    FieldSchema(
        name="id",
        dtype=DataType.INT64,
        is_primary=True,
        auto_id=False,
        max_length=100,
    ),
    FieldSchema(name="priority", dtype=DataType.FLOAT),
    FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR, dim=dim), ## dim = 100
]
schema = CollectionSchema(fields, "points with id and priority")
points = Collection("points", schema, consistency_level="Strong")


index_params = {
    "index_type": "DISKANN",
    "metric_type": "IP",
    "params": {},
}
points.create_index("embeddings", index_params)
print("done")
```

**Reply by yushangdi** [reply]
I tried again and found that I have to make indexNode.enableDisk to be `true` for it to work. Previously I was setting it to `false`. 

Does setting it to true mean the index will be stored on disk instead of in-memory? Is there any way to make the DiskANN index store fully in-memory?

**Reply by yhmo** [reply, marked as answer]
The default value of `indexNode.enableDisk` is true.
This configuration means "support disk index or not". True: support disk index. False: doesn't support disk index. 
Disk index requires hard disk. Sometimes, milvus is deployed on a host that has no hard disk. So we add this configuration item for this case.

Which kind of index is created depends on the index_params you pass to create_index().
FLAT/IVF?FLAT/HNSW/IVF_SQ8/IVF_PQ ---------------- fully in memory
DISKINDEX ------------------ partly in memory, partly on disk

Not possible to put DISKINDEX fully in memory.

**Reply by yushangdi** [reply]
I see. thanks!
